web/src/main/webapp/WEB-INF/pages/application_state.jsp

```diff
@@ -106,53 +133,156 @@
+                                style="display:none;">
+                                  <option disabled <c:if test="${data.recommendedPositionLevel eq null}"> selected="selected" </c:if>>
+                                                                          <spring:message code="common.pleaseChoose"/></option>
+                                  <c:forEach begin="0" end="5" step="1" var="index">
+                                      <option value="${index}" <c:if
+                                              test="${index eq data.recommendedPositionLevel}"> selected="selected" </c:if>>${index}</option>
+                                  </c:forEach>
+                              </select>
+                              </select>
+                          </c:if>
+                        </div>
+                        <div class="help-block with-errors"></div>
+                        </div>
                   </c:when>
+
                   <c:when test="${data.stateName == 'wageOffer'}">
                       <div class="form-group">
                           <label for="name" class="control-label col-sm-4"><spring:message code="statehistory.field.offeredMoney"/></label>
                           <div class="col-sm-8">
                               <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="offeredMoneyP">${data.offeredMoney}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="offeredMoney" id="offeredMoneyInput" style="display:none" value="${data.offeredMoney}">
+                                  <input class="stateInput" type="number" name="offeredMoney" id="offeredMoneyInput" style="display:none" value="${data.offeredMoney}"
+                                  data-error="<spring:message code="statehistory.error.offeredMoney.negative"/>"
+                                  data-bind="valueWithInit: 'offeredMoney'"
+                                  min="0">
                               </c:if>
                           </div>
+                          <div class="help-block with-errors"></div>
                       </div>
                       <div class="form-group">
                           <label for="name" class="control-label col-sm-4"><spring:message code="statehistory.field.claim"/></label>
                           <div class="col-sm-8">
                               <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="claimP">${data.claim}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="claim" id="claimInput" style="display:none" value="${data.claim}">
+                                  <input class="stateInput" type="number" name="claim" id="claimInput" style="display:none" value="${data.claim}"
```

Comment:
Use hidden class as described a comment below.

Also remove all occurrences of style="display:none" in your changes.